Services/Dashboard/Block/classes/class.ilRecommendedContentBlockGUI.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+class ilRecommendedContentBlockGUI extends ilDashboardBlockGUI
+{
+    public function initViewSettings(): void
+    {
+        $this->viewSettings = new ilPDSelectedItemsBlockViewSettings(
+            $this->user,
+            ilPDSelectedItemsBlockConstants::VIEW_RECOMMENDED_CONTENT
+        );
+
+        $this->ctrl->setParameter($this, 'view', $this->viewSettings->getCurrentView());
+    }
+
+    public function emptyHandling(): string
+    {
+        return '';
+    }
+
+    public function initData(): void
+    {
+        $rec_manager = new ilRecommendedContentManager();
+        $recommendations = $rec_manager->getOpenRecommendationsOfUser($this->user->getId());
+
+        $short_desc = $this->settings->get("rep_shorten_description");
+        $short_desc_max_length = (int) $this->settings->get("rep_shorten_description_length");
+        $ctrl = $this->ctrl;
+
+        $recommendations = array_map(static function ($ref_id) use ($short_desc, $short_desc_max_length) {
+            $obj_id = ilObject::_lookupObjectId($ref_id);
+            $desc = ilObject::_lookupDescription($obj_id);
+            if ($short_desc && $short_desc_max_length !== 0) {
+                $desc = ilStr::shortenTextExtended($desc, $short_desc_max_length, true);
+            }
+
+            return [
+               'title' => ilObject::_lookupTitle($obj_id),
+               'description' => $desc,
+               'ref_id' => $ref_id,
+               'obj_id' => $obj_id,
+               'url' => '',
+               'type' => ilObject::_lookupType($obj_id),
+            ];
+        }, $recommendations);
+
+        $this->setData(['' => $recommendations]);
+    }
+
+    public function getItemForData(array $data): ?\ILIAS\UI\Component\Item\Item
+    {
+        $item = $data;
+        /** @var ilObjectListGUI $itemListGui */
+        $list_factory = new ilPDSelectedItemsBlockListGUIFactory($this, $this->blockView);
+        $item_gui = $list_factory->byType($data['type']);
+        ilObjectActivation::addListGUIActivationProperty($item_gui, $item);
+
+        $this->ctrl->setParameterByClass(self::class, "item_ref_id", $data['ref_id']);
```

Comment:
Ich binn mir nicht ganz sicher ob du hier self::class verwenden willst.

Generell gibt es ja 3 methoden um an die Klasse zu kommen: self::class , __CLASS__ und getclass($this)

Der unterschied der ersten beiden zur dritten ist das get class auch in einer vererbung funktioniert.
Momentan ist das also kein Problem so, aber du muss entscheiden was du möchtest wenn andere Entwickler in Zukunft von deiner Klasse erben: Soll dort dann deine Klasse stehen oder ihre?

Dazu steht ihr auch noch mehr https://www.php.de/forum/webentwicklung/php-einsteiger/89234-self-vs-__class__